src/middlewares/authenticate.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response, NextFunction } from "express";
+import { User } from "../models/user";
+
+export const checkDuplicateEmail = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const user = await User.count({ where: { login: req.body.login } });
+    if (user) {
+      res.status(400).json({ error: "Email is already taken." });
```

Comment:
It's a good practice to keep all HTTP status codes in a separate config/constant module.